shell/platform/android/test/io/flutter/embedding/engine/renderer/FlutterEngineRule.java

```diff
@@ -0,0 +1,84 @@
+package io.flutter.embedding.engine.renderer;
+
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+import android.content.Context;
+import android.content.Intent;
+import androidx.test.core.app.ApplicationProvider;
+import io.flutter.embedding.android.FlutterActivity;
+import io.flutter.embedding.engine.FlutterEngine;
+import io.flutter.embedding.engine.FlutterEngineCache;
+import io.flutter.embedding.engine.FlutterJNI;
+import io.flutter.embedding.engine.loader.FlutterLoader;
+import org.junit.rules.TestWatcher;
+import org.junit.runner.Description;
+
+/**
+ * Prepares and returns a {@link FlutterEngine} and {@link Intent} primed with an engine for tests.
+ */
+public final class FlutterEngineRule extends TestWatcher {
+  private static final Context ctx = ApplicationProvider.getApplicationContext();
```

Comment:
Error: Do not place Android context classes in static fields; this is a memory leak [StaticFieldLeak]
  private static final Context ctx = ApplicationProvider.getApplicationContext();

Reply:
Thanks! Done.